src/site/twitch.tv/modules/chat/components/message/UserMessage.vue

```diff
@@ -251,6 +255,38 @@ watchEffect(() => {
 		).replace(/ (A|P)M/, "");
 	}
 });
+
+function isUserFocused() {
+	if (!focusedChatters) return false;
+	return focusedChatters.value.includes(msg.value.author!.username);
+}
+
+function onUsernameDoubleClick() {
+	if (!focusedChatters) return;
+	focusedChatters.value.push(msg.value.author!.username);
+
+	// Clear blue selection after double clicking username
+	document.getSelection()?.empty();
+}
+
+let clickTimer = -1;
+function handleUsernameClick(ev: MouseEvent) {
+	if (clickToFocusOnChatterEnabled.value) {
+		const target = ev.currentTarget;
+		if (ev.detail === 1) {
+			clickTimer = setTimeout(() => {
+				// Single click
+				openViewerCard(ev, msg.value.author!.username, msg.value.id, target);
+			}, 300);
+		} else if (ev.detail >= 2) {
+			// Double click
+			clearTimeout(clickTimer);
+			onUsernameDoubleClick();
+		}
+	} else {
+		openViewerCard(ev, msg.value.author!.username, msg.value.id);
+	}
+}
```

Comment:
Such logic should be built into a [directive](https://vuejs.org/guide/reusability/custom-directives.html).